### PR TITLE
r/s3_bucket_lifecycle_configuration: correctly configure `filter` `tag` and add `and` block documentation

### PR DESCRIPTION
**Pull request by anGie44**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes https://github.com/hashicorp/terraform-provider-aws/issues/23239

Output from acceptance testing:

```
--- PASS: TestAccS3BucketLifecycleConfiguration_Filter_Tag (193.92s)

--- PASS: TestAccS3BucketLifecycleConfiguration_TransitionZeroDays_intelligentTiering (197.49s)
--- PASS: TestAccS3BucketLifecycleConfiguration_TransitionStorageClassOnly_intelligentTiering (219.40s)
--- PASS: TestAccS3BucketLifecycleConfiguration_RuleExpiration_expireMarkerOnly (281.11s)
--- PASS: TestAccS3BucketLifecycleConfiguration_TransitionUpdateBetweenDaysAndDate_intelligentTiering (332.06s)
--- PASS: TestAccS3BucketLifecycleConfiguration_TransitionDate_standardIa (193.00s)
--- PASS: TestAccS3BucketLifecycleConfiguration_TransitionDate_intelligentTiering (193.25s)
--- PASS: TestAccS3BucketLifecycleConfiguration_RuleExpiration_emptyBlock (193.54s)
--- PASS: TestAccS3BucketLifecycleConfiguration_ruleAbortIncompleteMultipartUpload (273.25s)
--- PASS: TestAccS3BucketLifecycleConfiguration_nonCurrentVersionExpiration (194.30s)
--- PASS: TestAccS3BucketLifecycleConfiguration_prefix (203.91s)
--- PASS: TestAccS3BucketLifecycleConfiguration_disappears (54.16s)
--- PASS: TestAccS3BucketLifecycleConfiguration_nonCurrentVersionTransition (193.61s)
--- PASS: TestAccS3BucketLifecycleConfiguration_multipleRules (193.32s)
--- PASS: TestAccS3BucketLifecycleConfiguration_disableRule (327.59s)
--- PASS: TestAccS3BucketLifecycleConfiguration_filterWithPrefix (272.06s)
```

### CODE REVIEW

File: website/docs/r/s3_bucket_lifecycle_configuration.html.markdown [hunk 1]
```diff
@@ -208,6 +190,15 @@ The `transition` configuration block supports the following arguments:
 * `days` - (Optional, Conflicts with `date`) The number of days after creation when objects are transitioned to the specified storage class. The value must be a positive integer. If both `days` and `date` are not specified, defaults to `0`. Valid values depend on `storage_class`, see [Transition objects using Amazon S3 Lifecycle](https://docs.aws.amazon.com/AmazonS3/latest/userguide/lifecycle-transition-general-considerations.html) for more details.
 * `storage_class` - The class of storage used to store the object. Valid Values: `GLACIER`, `STANDARD_IA`, `ONEZONE_IA`, `INTELLIGENT_TIERING`, `DEEP_ARCHIVE`, `GLACIER_IR`.
 
+### and
```
Comment (YakDriver) [inline website/docs/r/s3_bucket_lifecycle_configuration.html.markdown:193]: I like the upgrade guide having `filter` examples using `and` and just `prefix`.

**Review by YakDriver — APPROVED** [review]
MPB 🎉 

```console
% make testacc TESTS=TestAccS3BucketLifecycle PKG=s3 ACCTEST_PARALLELISM=5
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/s3/... -v -count 1 -parallel 5 -run='TestAccS3BucketLifecycle'  -timeout 180m
--- PASS: TestAccS3BucketLifecycleConfiguration_basic (198.58s)
--- PASS: TestAccS3BucketLifecycleConfiguration_TransitionZeroDays_intelligentTiering (198.96s)
--- PASS: TestAccS3BucketLifecycleConfiguration_TransitionStorageClassOnly_intelligentTiering (199.20s)
--- PASS: TestAccS3BucketLifecycleConfiguration_RuleExpiration_expireMarkerOnly (279.70s)
--- PASS: TestAccS3BucketLifecycleConfiguration_TransitionUpdateBetweenDaysAndDate_intelligentTiering (335.33s)
--- PASS: TestAccS3BucketLifecycleConfiguration_TransitionDate_standardIa (195.21s)
--- PASS: TestAccS3BucketLifecycleConfiguration_TransitionDate_intelligentTiering (195.65s)
--- PASS: TestAccS3BucketLifecycleConfiguration_RuleExpiration_emptyBlock (194.92s)
--- PASS: TestAccS3BucketLifecycleConfiguration_ruleAbortIncompleteMultipartUpload (275.88s)
--- PASS: TestAccS3BucketLifecycleConfiguration_disappears (54.79s)
--- PASS: TestAccS3BucketLifecycleConfiguration_multipleRules (195.78s)
--- PASS: TestAccS3BucketLifecycleConfiguration_prefix (192.20s)
--- PASS: TestAccS3BucketLifecycleConfiguration_filterWithPrefix (276.06s)
--- PASS: TestAccS3BucketLifecycleConfiguration_disableRule (340.00s)
--- PASS: TestAccS3BucketLifecycleConfiguration_Filter_Tag (203.28s)
--- PASS: TestAccS3BucketLifecycleConfiguration_nonCurrentVersionTransition (191.94s)
--- PASS: TestAccS3BucketLifecycleConfiguration_nonCurrentVersionExpiration (192.04s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/s3	861.198s
```

**Comment by Tenzer** [comment]
The bug reported in #23132 was for version 4.0.0, was fixed in 4.1.0 and is now a problem again in 4.2.0, likely due to the change in this PR. Could the bug be reopened again, please?

**Comment by anGie44** [comment]
Hi @Tenzer 👋 , please see my latest comment in https://github.com/hashicorp/terraform-provider-aws/issues/23132 . based on your reply there, we can further evaluate next steps.

**Pull request merged**